aws/resource_aws_codeartifact_domain.go

```diff
@@ -0,0 +1,122 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/codeartifact"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceAwsCodeArtifactDomain() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCodeArtifactDomainCreate,
+		Read:   resourceAwsCodeArtifactDomainRead,
+		Delete: resourceAwsCodeArtifactDomainDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"domain": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"encryption_key": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
```

Comment:
Validate with `validateArn` to ensure that we don't have to worry about diffs with key IDs and aliases being used?